packages/batch-submitter/test/batch-submitter/mockchain-provider.ts

```diff
@@ -60,25 +61,32 @@ export class MockchainProvider extends providers.JsonRpcProvider {
   }
 
   public async send(endpoint: string, params: []): Promise<any> {
-    if (endpoint === 'eth_chainId') {
-      return this.chainId()
+    switch (endpoint) {
+      case 'eth_chainId':
+        return this.chainId()
+      case 'rollup_getInfo':
+        const info: RollupInfo = {
+          mode: 'sequencer',
+          syncing: false,
+          ethContext: {
+            timestamp: 0,
+            blockNumber: 0,
+          },
+          rollupContext: {
+            index: 0,
+            queueIndex: 0,
+          },
+        }
+        return info
+      case 'eth_getBlockByNumber':
+        if (params.length === 0) {
+          throw new Error(`Invalid params for ${endpoint}`)
+        }
+        const blockNumber = BigNumber.from((params as any)[0]).toNumber()
+        return this.mockBlocks[blockNumber]
```

Comment:
we might be able to remove this if `injectL2Context` lets us call `getBlockWithTransactions`

Reply:
Mockchain provider refactor is going to be a beast -- thinking that we should just lump that into the big testing overhaul

Reply:
noted!